### PR TITLE
Unify JS/TS file naming [MAILPOET-4938]

### PR DESCRIPTION
**Pull request by JanJakes**
## Description

I used a simple script to do the file and import renames, the rest was done manually. See commit messages for more info.

## Code review notes

Is there anything more that could reference some JS files that I forgot to change? I went through all configs and stuff and most cases should be caught by our CI.

## QA notes

Please, perform a general smoke test. Ensure that all plugin pages load, etc.

## Linked PRs

https://github.com/mailpoet/mailpoet-premium/pull/792

## Linked tickets

[MAILPOET-4938]

## After-merge notes

Ping @JanJakes, please, to notify others about this change.

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-4938]: https://mailpoet.atlassian.net/browse/MAILPOET-4938?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/package.json [hunk 1]
```diff
@@ -75,6 +75,7 @@
     "codemirror": "^5.65.2",
     "css": "^3.0.0",
     "date-fns": "^2.28.0",
+    "eslint-plugin-check-file": "^2.6.2",
```
Comment (brezocordero) [inline mailpoet/package.json]: Why not a dev dependency?

**Review by brezocordero — DISMISSED** [review]
LGTM.
I have done some random checks + a bit of grep + verify config files
Did not stop anything you could have missed.

Only one question about the plugin `eslint-plugin-check-file`

File: mailpoet/package.json (hunk 1 above)
Comment (JanJakes) [reply mailpoet/package.json]: @brezocordero Thanks for the catch, I only forgot to add the dev flag. Fixed & rebased.

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**